### PR TITLE
controller util SortByCreationTimestamp clean up

### PR DESCRIPTION
**Pull request by mqliang**
@kargakis

### CODE REVIEW

File: pkg/controller/controller_utils.go [hunk 1]
```diff
@@ -466,26 +466,26 @@ func SyncAllPodsWithStore(kubeClient clientset.Interface, store cache.Store) {
 	return
 }
 
-// ControllersByCreationTimestamp sorts a list of ReplicationControllers by creation timestamp, using their names as a tie breaker.
```
Comment (kargakis) [inline pkg/controller/controller_utils.go]: Leave the rc related type around for other callers. It should be removed until all rc related code is removed.

File: pkg/controller/replication/replication_controller.go [hunk 2]
```diff
@@ -216,7 +216,7 @@ func (rm *ReplicationManager) getPodController(pod *api.Pod) *api.ReplicationCon
 		// overlap, sort by creation timestamp, subsort by name, then pick
 		// the first.
 		glog.Errorf("user error! more than one replication controller is selecting pods with labels: %+v", pod.Labels)
-		sort.Sort(OverlappingControllers(controllers))
+		sort.Sort(overlappingControllers(controllers))
```
Comment (kargakis) [inline pkg/controller/replication/replication_controller.go]: remove `OverlappingControllers` and use `ControllersByCreationTimestamp`

**Comment by madhusudancs** [comment]
In addition to @kargakis comments, please also remove overlappingReplicaSets from pkg/controller/replicaset/replica_set_utils.go and refactor pkg/controller/replicaset/replica_set.go to use the sortable type in pkg/controller/controller_utils.go.

**Comment by madhusudancs** [comment]
Also, a note FYI because I am seeing a few PRs flying by trying to replace Replication Controller with Replica Set. While we want Replica Set to be the next-gen Replication Controller, we want to neither actively remove Replication Controller from the code nor replace every occurrence with Replica Set at this time. Deployments was switched to Replica Set because of two reasons:
1. We want new types (example: controllers) with selectors to support the new label selector format, i.e. the set-based selector requirements as described here - https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/labels.md#set-based-requirement. We couldn't do that for Deployments because Deployments used Replication Controllers and RCs only supported the old map[string]string type selectors. It was difficult to introduce the new selector semantics with RC in a backwards-compatible way. See https://github.com/kubernetes/kubernetes/issues/341#issuecomment-140809259. So we implemented Replica Set, which today is merely a replica (pun intended ;-)) of Replication Controllers except for the selector and switched Deployments to Replica Set so that we could support the new selector format.
2. Deployments was still in alpha.

So we shouldn't remove/replace all RC occurrences with Replica Sets.

**Comment by mqliang** [comment]
@madhusudancs 

I think I got it: We can replace ReplicationController to ReplicaSet in DeploymentController because k8s v1.0 has no DeploymentController. But for selector_spreading functionality in scheduler, we must make it back compatible with v1.0. Hence, we should ADD ReplicaSet, and keep ReplicationController. Thanks for your hint.

File: pkg/controller/replication/replication_controller.go (hunk 2 above)
Comment (mqliang) [inline pkg/controller/replication/replication_controller.go]: May be it's not easy: `ControllersByCreationTimestamp` sorts an array of RC pointers, by `OverlappingControllers ` sorts an array of RCs

Comment (kargakis) [inline pkg/controller/replication/replication_controller.go]: then pass a reference to controllers here

Comment (kargakis) [inline pkg/controller/replication/replication_controller.go]: ah I see, nevermind

**Comment by bgrant0607** [comment]
cc @bprashanth

**Pull request closed without merging**